src/osp/Active/SysNewton.cpp

```diff
@@ -571,29 +594,42 @@ Matrix3 SysNewton::compute_mass_inertia(ActiveScene& rScene, ActiveEnt part)
 
         Matrix4 childMatrix = childTransform.m_transform;
 
+        auto* childCollide = rScene.reg_try_get<ACompCollisionShape>(nextChild);
+        auto* childContainer = rScene.reg_try_get<ACompContainerShape>(nextChild);
+
         // If entity has a collider, add its volume and inertia to sum
-        if (auto* childCollide = rScene.reg_try_get<ACompCollisionShape>(nextChild);
-            childCollide != nullptr)
+        if (childCollide || childContainer)
         {
-            float volume = phys::col_shape_volume(childCollide->m_shape, childMatrix.scaling());
-            float mass = volume * partDensity;
+            float mass = 0.0f;
+            phys::ECollisionShape shape;
+
+            if (childCollide)
+            {
+                float volume =
+                    phys::col_shape_volume(childCollide->m_shape, childMatrix.scaling());
+
+                mass = volume * partDensity;
+                shape = childCollide->m_shape;
+                
+            }
+            else if (childContainer)
+            {
+                auto& containerMass = rScene.reg_get<ACompMass>(nextChild);
+
+                mass = containerMass.m_mass;
+                shape = childContainer->m_shape;
+            }
```

Comment:
The annoying thing about this arrangement is that different types of masses need to be computed differently; perhaps part densities should be precalculated and assigned to ACompMass entities like I've done with the containers? That would allow us to avoid coupling the details of the mass distribution to Newton, especially if we had some way to unify the shape representation.